userspace/libsinsp/k8s_handler.cpp

```diff
@@ -410,7 +411,23 @@ void k8s_handler::collect_data()
 				send_data_request();
 				if(m_blocking_socket && !m_watching)
 				{
-					receive_response();
+					try
+					{
+						receive_response();
+					}
+					catch(const sinsp_exception& ex)
```

Comment:
Is a `sinsp_exception& ex` enough here, or maybe we also need something stronger like `catch(...)` ?

Reply:
I think it is enough, since we are throwing only `sinsp_exception` if something goes wrong in our code.